trpcultivate_germplasm/tests/src/Kernel/GermplasmTermInstallTest.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+namespace Drupal\Tests\trpcultivate_germplasm\Kernel;
+
+use Drupal\Tests\tripal_chado\Kernel\ChadoTestKernelBase;
+use Drupal\Tests\trpcultivate_germplasm\Traits\TripalMviewQueriesTestTrait;
+use Drupal\tripal_chado\Database\ChadoConnection;
+use Drupal\user\Entity\User;
+
+/**
+ * Tests term install by the trpcultivate_germplasm.module.
+ *
+ * @group trpcultivate_germplasm
+ */
+class GermplasmTermInstallTest extends ChadoTestKernelBase {
+
+  /**
+   * Test Trait for setting up necessary materialized views in Drupal.
+   */
+  use TripalMviewQueriesTestTrait;
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  protected static $modules = [
+    'system',
+    'file',
+    'user',
+    'path_alias',
+    'tripal',
+    'tripal_chado',
+    'trpcultivate_germplasm',
+  ];
+
+  /**
+   * A Database query interface for querying Chado using Tripal DBX.
+   *
+   * @var \Drupal\tripal_chado\Database\ChadoConnection
+   */
+  protected ChadoConnection $chado_connection;
+
+  /**
+   * A Database query interface for querying Drupal database.
+   *
+   * @var \Drupal\core\Database\Database
+   */
+  protected $drupal_connection;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Set test environment.
+    \Drupal::state()->set('is_a_test_environment', TRUE);
+
+    // Create a test chado instance as needed by our service.
+    $this->chado_connection = $this->createTestSchema(ChadoTestKernelBase::PREPARE_TEST_CHADO);
+
+    // Grab our drupal connection.
+    $this->drupal_connection = $this->container->get('database');
+
+    // Setup test evironment.
+    $this->prepareEnvironment(['TripalTerm']);
+
+    $this->installSchema('tripal', ['tripal_import', 'tripal_jobs']);
+    $this->installSchema('tripal_chado', [
+      'tripal_cv_obo',
+      'tripal_custom_tables',
+      'tripal_mviews',
+    ]);
+
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('path_alias');
+
+    $this->installConfig([
+      'system',
+      'trpcultivate_germplasm',
+    ]);
+
+    $this->container->get('module_installer')->install([
+      'tripal',
+      'tripal_chado',
+      'trpcultivate_germplasm',
+    ]);
+
+    // Create test user.
+    $user = User::create([
+      'name' => 'user-collector',
+      'roles' => ['authenticated user'],
+    ]);
+    $user->save();
+
+    $this->container->get('current_user')
+      ->setAccount($user);
+
+    // Set up our necessary materialized views.
```

Comment:
```suggestion
    // Set up our necessary materialized views.
    // This is needed because Tripal Core does not yet setup the Drupal side of  
    // the test environment correctly when it comes to materialized views and  
    // custom tables. Hopefully that will be resolved in the future.
```